src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HPack/HPackEncoder.cs

```diff
@@ -24,61 +24,110 @@ internal static class HPackEncoder
 
         public static bool EncodeHeader(string name, string value, Span<byte> buffer, out int length)
         {
-            Debug.Assert(value != null);
-            return EncodeHeader(name, value, null, null, buffer, out length);
-        }
+            int i = 0;
+            length = 0;
+            int valueLength = (value == null ? 0 : value.Length);
```

Comment:
Can value actually be null here, or should it be an empty string instead?

Reply:
I was not sure. But since we have test now and it does get empty value I updated this line.